cpp/arcticdb/column_store/column.hpp

```diff
@@ -676,21 +688,16 @@ template <typename T>
 JiveTable create_jive_table(const Column& col) {
     JiveTable output(col.row_count());
     std::iota(std::begin(output.orig_pos_), std::end(output.orig_pos_), 0);
-    std::iota(std::begin(output.sorted_pos_), std::end(output.sorted_pos_), 0);
 
+    // Calls to scalar_at are expensive, so we precompute them to speed up the sort compare function.
+    auto values = col.template clone_scalars_to_vector<T>();
```

Comment:
The thing that takes the time in scalar_at is the branching each time where we work out whether there's one contiguous block, a bunch of regular-sized ones of irregulat blocks. I think what we want to do eventually is return a variant column iterator that we can visit on which in the contiguous case just acts like a pointer + offset, in the regular case does a mask and only does the complicated stuff where blocks are irregular-sized (which is quite unusual). @JohanMabille will be looking at this sort of issue in due course, so I think this is fine for the moment

Reply:
Agreed, if we have a fast random access iterator on the column data cloning to a vector will become obsolete. Should I return to the scalar_at calls now? (producing a significant slowdown (I think around 4x) which will be fixed later on with Johan's smarter iterators)

Reply:
No, I think leave it as it is for the moment, we'll just have to remember to go back and fix it later